packages/fs-kernels/src/kernel.ts

```
@@ -26,14 +34,67 @@ export class Kernel {
     this.channels = launchedKernel.channels;
   }
 
-  async shutdown() {
+  shutdownEpic(timeoutMs: number = 2000) {
+    const request = shutdownRequest({ restart: false });
+
+    // Try to make a shutdown request
+    // If we don't get a response within X time, force a shutdown
+    // Either way do the same cleanup
+    const shutDownHandling = this.channels.pipe(
+      /* Get the first response to our message request. */
+      childOf(request),
+      ofMessageType("shutdown_reply"),
+      first(),
+      // If we got a reply, great! :)
+      map((msg: { content: { restart: boolean } }) => {
+        return {
+          status: "shutting down",
+          content: msg.content
+        };
+      }),
+      /**
+       * If we don't get a response within timeoutMs, then throw an error.
+       */
+      timeout(timeoutMs),
+      catchError(err => of({ error: err })),
+      /**
+       * Even if we don't receive a shutdown_reply from the kernel to our
+       * shutdown_request, we will go forward with cleaning up the RxJS
+       * subject and killing the kernel process.
+       */
+      mergeMap(async () => {
```

In this function we should probably propagate the error from above, even though we're also going to force shut down fully. Something like:

```
mergeMap(async (event) => {
  this.channels.complete();
  await this.shutdownProcess();

  const finalResponse = {status: "shutdown"};
  if(message.error) {
    finalResponse.error = message.error;
  }
  return of(finalResponse);
}); 
```

Not the cleanest, but does provide transparency for failure here.